src/sentry_plugins/sessionstack/static/sessionstack/contexts/sessionstack.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+const ASPECT_RATIO = 16 / 9;
+
+const SessionStackContextType = React.createClass({
+  propTypes: {
+    alias: React.PropTypes.string.isRequired,
+    data: React.PropTypes.object.isRequired
+  },
+
+  getInitialState() {
+    return {
+      showIframe: false
+    };
+  },
+
+  componentDidMount() {
+    this.parentNode = ReactDOM.findDOMNode(this).parentNode;
+    window.addEventListener('resize', this.setIframeSize, false);
+    this.setIframeSize();
+  },
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.setIframeSize, false);
+  },
+
+  setIframeSize() {
+    if (!this.showIframe) {
+      let parentWidth = $(this.parentNode).width();
+
+      this.setState({
+        width: parentWidth,
+        height: parentWidth / ASPECT_RATIO
+      });
+    }
+  },
+
+  playSession() {
+    this.setState({
+      showIframe: true
+    });
+
+    this.setIframeSize();
+  },
+
+  render() {
+    let { session_url } = this.props.data;
+
+    if (!session_url) {
+      return (
+        <h4>Session not found.</h4>
+      );
+    }
+
+    return (
+      <div className="panel-group">
+        {this.state.showIframe ?
+          <iframe src={session_url} width={this.state.width} height={this.state.height} /> :
```

Comment:
>  I think it'd have access to the parent frame and be able to execute JavaScript. I could be wrong here, but our CSP rules would reject it anyways.

It can't. But it can redirect the parent page via `window.parent.location`, which can be used in a phishing attack (e.g. redirect to a page that looks like a Sentry login, attempt to trick user into divulging password).

I think we definitely need to lock down the potential iframe URLs. And I think we should explore enforcing CSP on the iframe itself (e.g. the iframe cannot be allowed to execute unknown, arbitrary scripts).